mobile/src/main/java/org/openhab/habdroid/model/Widget.kt

```diff
@@ -214,13 +214,15 @@ fun String?.toWidgetType(): Widget.Type {
     return Widget.Type.Unknown
 }
 
-fun String?.toInputHint(): Widget.InputTypeHint {
+fun String?.toInputHint(item: Item?): Widget.InputTypeHint {
     if (this != null) {
         try {
             return Widget.InputTypeHint.valueOf(this.toString().lowercase().replaceFirstChar { c -> c.uppercase() })
         } catch (e: IllegalArgumentException) {
             // fall through
         }
+    } else {
+        if (item?.isOfTypeOrGroupType(Item.Type.DateTime) == true) return Widget.InputTypeHint.Datetime
```

Comment:
I would rather do this when consuming the input hint. IMO conversion methods shouldn't introduce policy.

Reply:
@maniac103 I have changed it, please check my implementation is in line with what you had in mind here.

Reply:
It roughly is what I had in mind ;-) I'd do this: https://github.com/maniac103/openhab.android/commit/ac93cd67d3cd7e915b2879d4818f7a928a224588